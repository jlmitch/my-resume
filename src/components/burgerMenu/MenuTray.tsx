import React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';

interface MenuTrayProps {
  open: boolean;
  pathname: string;
}

const Ul = styled.ul<Omit<MenuTrayProps, 'pathname'>>`
  background-color: #BFA181;
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
  list-style: none;
  padding-top: 3.5rem;
  position: fixed;
  right: 0;
  top: -16px;
  transform: ${({ open }) => open ? 'translateX(0)' : 'translateX(100%)'};
  transition: transform 0.3s ease-in-out;
  width: 300px;

  & .nav-link-item {
    padding: 18px 10px;
    & .nav-link {
      color: #178582;
      text-decoration: none;
      text-transform: uppercase;
      font-weight: 700;

      &:hover {
        border-bottom: solid 2px #0A1828;
      }
    }
  }
`;


const MenuTray = ({ open, pathname }: MenuTrayProps) => {
  return (
    <Ul open={open}>
      <li className="nav-link-item">
        <Link className={`nav-link ${pathname === '/' ? 'active' : '' }`} to="/"><span>Home</span></Link>
      </li>
      <li className="nav-link-item">
        <Link className={`nav-link ${pathname === '/about/' ? 'active' : '' }`} to="/about"><span>About</span></Link>
      </li>
      <li className="nav-link-item">
        <Link className={`nav-link ${pathname === '/resume/' ? 'active' : '' }`} to="/resume"><span>Resume</span></Link>
      </li>
      <li className="nav-link-item">
        <Link className={`nav-link ${pathname === '/blog/' ? 'active' : '' }`} to="/blog"><span>Blog</span></Link>
      </li>
    </Ul>
  )
}

export default MenuTray
