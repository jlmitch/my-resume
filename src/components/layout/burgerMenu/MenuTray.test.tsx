import React from 'react';
import { render, screen } from '@testing-library/react';
import MenuTray from './MenuTray';

describe('MenuTray', () => {
  test('should display MenuTray content', () => {
    const { baseElement } = render(<MenuTray open={true} pathname="/cats/" />);

    expect(baseElement).toMatchSnapshot();
  });
});
