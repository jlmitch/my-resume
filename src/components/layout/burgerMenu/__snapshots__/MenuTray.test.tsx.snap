// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuTray should display MenuTray content 1`] = `
<body>
  <div>
    <ul
      class="sc-beqWaB iqOwOs"
      open=""
    >
      <li
        class="nav-link-item"
      >
        <a
          class="nav-link "
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="nav-link-item"
      >
        <a
          class="nav-link "
          href="/about"
        >
          <span>
            About
          </span>
        </a>
      </li>
      <li
        class="nav-link-item"
      >
        <a
          class="nav-link "
          href="/resume"
        >
          <span>
            Resume
          </span>
        </a>
      </li>
    </ul>
  </div>
</body>
`;
